src/cmd/dev.go

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023-Present The UDS Authors
+
+// Package cmd contains the CLI commands for UDS.
+package cmd
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	tea "github.com/charmbracelet/bubbletea"
+
+	"github.com/defenseunicorns/uds-cli/src/config"
+	"github.com/defenseunicorns/uds-cli/src/config/lang"
+	"github.com/defenseunicorns/uds-cli/src/pkg/bundle"
+	"github.com/defenseunicorns/uds-cli/src/pkg/bundle/tui/deploy"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/utils/helpers"
+	"github.com/spf13/cobra"
+	"golang.org/x/term"
+)
+
+var devCmd = &cobra.Command{
+	Use:   "dev",
+	Short: lang.CmdDevShort,
+}
+
+var devDeployCmd = &cobra.Command{
+	Use:   "deploy",
+	Args:  cobra.MaximumNArgs(1),
+	Short: lang.CmdDevDeployShort,
+	PreRun: func(_ *cobra.Command, args []string) {
+		pathToBundleFile := ""
+		if len(args) > 0 {
+			if !helpers.IsDir(args[0]) {
+				message.Fatalf(nil, "(%q) is not a valid path to a directory", args[0])
+			}
+			pathToBundleFile = filepath.Join(args[0])
+		}
+		// Handle .yaml or .yml
+		bundleYml := strings.Replace(config.BundleYAML, ".yaml", ".yml", 1)
+		if _, err := os.Stat(filepath.Join(pathToBundleFile, config.BundleYAML)); err == nil {
+			bundleCfg.CreateOpts.BundleFile = config.BundleYAML
+		} else if _, err = os.Stat(filepath.Join(pathToBundleFile, bundleYml)); err == nil {
+			bundleCfg.CreateOpts.BundleFile = bundleYml
+		} else {
+			message.Fatalf(err, "Neither %s or %s found", config.BundleYAML, bundleYml)
+		}
+	},
+	Run: func(_ *cobra.Command, args []string) {
+
+		// Create Bundle
+		srcDir, err := os.Getwd()
+		if err != nil {
+			message.Fatalf(err, "error reading the current working directory")
+		}
+		if len(args) > 0 {
+			srcDir = args[0]
+		}
+
+		config.CommonOptions.Confirm = true
+		bundleCfg.CreateOpts.SourceDirectory = srcDir
+		configureZarf()
+
+		// load uds-config if it exists
+		if v.ConfigFileUsed() != "" {
+			if err := loadViperConfig(); err != nil {
+				message.Fatalf(err, "Failed to load uds-config: %s", err.Error())
+				return
+			}
+		}
+
+		bndlClient := bundle.NewOrDie(&bundleCfg)
+		defer bndlClient.ClearPaths()
+
+		// Check if local zarf packages need to be created
+		if len(srcDir) != 0 && srcDir[len(srcDir)-1] != '/' {
+			srcDir = srcDir + "/"
+		}
+		bndlClient.CreateZarfPkgs(srcDir)
+
+		// Create dev bundle
+		config.Dev = true
+		if err := bndlClient.Create(); err != nil {
```

Comment:
What do you think of having the created bundle have a slightly different filename to indicate that it's dev bundle, something like: `uds-bundle-test-local-and-remote-arm64-0.0.1-DEV.tar.zst`. Just trying to avoid accidentally publishing a dev bundle instead of a non-dev one

Reply:
ahhh gotcha, didn't realize the bundles were identical. That makes sense because you're modifying the pkg in memory